src/pages/SearchPage.js

```diff
@@ -35,209 +32,193 @@ const propTypes = {
-    constructor(props) {
-        super(props);
+function SearchPage({betas, personalDetails, reports, isSearchingForReports}) {
+    // Data for initialization (runs only on the first render)
+    const {
+        recentReports: initialRecentReports,
+        personalDetails: initialPersonalDetails,
+        userToInvite: initialUserToInvite,
+        // Ignoring the rule because in this case we need the data only initially
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    } = useMemo(() => OptionsListUtils.getSearchOptions(reports, personalDetails, '', betas), []);
+
+    const [searchValue, setSearchValue] = useState('');
+    const [searchOptions, setSearchOptions] = useState({
+        recentReports: initialRecentReports,
+        personalDetails: initialPersonalDetails,
+        userToInvite: initialUserToInvite,
+    });
+
+    const {isOffline} = useNetwork();
+    const {translate} = useLocalize();
+    const themeStyles = useThemeStyles();
+    const isMounted = useRef(false);
+
+    const updateOptions = useCallback(() => {
+        const {
+            recentReports: localRecentReports,
+            personalDetails: localPersonalDetails,
+            userToInvite: localUserToInvite,
+        } = OptionsListUtils.getSearchOptions(reports, personalDetails, searchValue.trim(), betas);
+
+        setSearchOptions({
+            recentReports: localRecentReports,
+            personalDetails: localPersonalDetails,
+            userToInvite: localUserToInvite,
+        });
+    }, [reports, personalDetails, searchValue, betas]);
+
+    const debouncedUpdateOptions = useMemo(() => _.debounce(updateOptions, 75), [updateOptions]);
 
+    useEffect(() => {
         Timing.start(CONST.TIMING.SEARCH_RENDER);
         Performance.markStart(CONST.TIMING.SEARCH_RENDER);
+    }, []);
 
-        this.searchRendered = this.searchRendered.bind(this);
-        this.selectReport = this.selectReport.bind(this);
-        this.onChangeText = this.onChangeText.bind(this);
-        this.updateOptions = this.updateOptions.bind(this);
-        this.debouncedUpdateOptions = _.debounce(this.updateOptions.bind(this), 75);
-        this.state = {
-            searchValue: '',
-            recentReports: {},
-            personalDetails: {},
-            userToInvite: {},
-        };
-    }
-
-    componentDidUpdate(prevProps) {
-        if (_.isEqual(prevProps.reports, this.props.reports) && _.isEqual(prevProps.personalDetails, this.props.personalDetails)) {
+    useEffect(() => {
+        if (!isMounted.current) {
+            isMounted.current = true;
             return;
         }
-        this.updateOptions();
-    }
 
-    onChangeText(searchValue = '') {
-        Report.searchInServer(searchValue);
-        this.setState({searchValue}, this.debouncedUpdateOptions);
-    }
+        debouncedUpdateOptions();
+        // Ignoring the rule intentionally, we want to run the code only when search Value changes to prevent additional runs.
```

Comment:
Do we? They should update if `reports` or `personalDetails` change I think...

Reply:
No, I believe we want to fire the debounce function only on the searchValue change - in other cases we use updateOptions
<img width="1186" alt="Screenshot 2023-12-15 at 16 15 07" src="https://github.com/Expensify/App/assets/28020445/75706d68-1a17-40ff-b87c-cc62692a3959">

Reply:
Ok, thats sounds right. 

But the only place I see `updateOptions()` getting called is here: 

https://github.com/Expensify/App/blob/443a9cafca94355e0fd0db1a3eb4cffbe8dce85a/src/pages/SearchPage.js#L175

Which suggests we will not update them if the data changes only when the screen opens or user is typing.

Lemme know if I missed something though, thank you!

Reply:
<img width="941" alt="Screenshot 2023-12-18 at 08 18 31" src="https://github.com/Expensify/App/assets/28020445/3baed719-ce07-42c6-8a71-74bd5858f571">

updateOptions is also called inside the debouncedUpdateOptions and in its dependency array we have reports and personalDetails. I wonder how to actually test it 😅 (the data changing in other cases than screen being open and user typing) but to me the major functionality of this component is preserved and working well 🙂

Reply:
@marcaaron I think our discussion here got a little outdated 😃 - after this PR we no longer use debounced function https://github.com/Expensify/App/pull/32962

ready for re-review

Reply:
It is used here:

https://github.com/Expensify/App/blob/ad4cea8aac51e3cdc339eb6411bbb4676f0ce225/src/pages/SearchPage.js#L91-L94

And here:

https://github.com/Expensify/App/blob/ad4cea8aac51e3cdc339eb6411bbb4676f0ce225/src/pages/SearchPage.js#L171

>in its dependency array we have reports and personalDetails

Correct! We still must change the search value in order to update the options.

The logic is changing from what we have now:

https://github.com/Expensify/App/blob/4b950101272fd9c73c160f40c4fc889ebb4b7b55/src/pages/SearchPage.js#L79-L84

>the major functionality of this component is preserved and working well

If the data changes the options would not update. I have not tested yet, but I think this could be a problem because:

- User types / searches
- New results could appear from the server that are not available yet
- They must type again to see these results

Reply:
@marcaaron applied, but I do it now in another useEffect - I'm leaving the one with searchValue and debounce separate (we reverted the changes that were removing the denounced function)

Reply:
Thanks!